src/v3/src/transformer/button/transformIDPButtons.ts

```diff
@@ -10,93 +10,58 @@
 import {
-  ButtonElement,
-  ButtonType,
   DividerElement,
-  IWidgetContext,
   TransformStepFnWithOptions,
 } from '../../types';
-import { loc } from '../../util';
+import { getFastPassButtonElement, getIdpButtonElements, loc } from '../../util';
 
-// TODO: OKTA-504638 Implement CUSTOM IDP Buttons here
 export const transformIDPButtons: TransformStepFnWithOptions = ({
   transaction,
   widgetProps,
 }) => (
   formbag,
 ) => {
   const { neededToProceed: remediations } = transaction;
-  const containsPivIDP = remediations.some(
-    (remediation) => remediation.name === IDX_STEP.PIV_IDP,
-  );
   const containsIdentifyStep = remediations.some(
     (remediation) => remediation.name === IDX_STEP.IDENTIFY,
   );
-  // To display button must be on identify page and contain PIV remediation
-  if (!(containsPivIDP && containsIdentifyStep) ) {
+  const containsEnrollProfileStep = remediations.some(
+    (remediation) => remediation.name === IDX_STEP.ENROLL_PROFILE,
+  );
+
+  // must be on identify or enroll profile page to display Idp buttons
+  if (!(containsIdentifyStep || containsEnrollProfileStep)) {
     return formbag;
   }
 
-  const { piv, idpDisplay } = widgetProps;
-  const isBottomDivider = typeof idpDisplay === 'undefined'
-    ? true
-    : idpDisplay.toUpperCase() === 'PRIMARY';
-  const pivButton: ButtonElement = {
-    type: 'Button',
-    label: piv?.text || loc('piv.cac.card', 'login'),
-    options: {
-      type: ButtonType.BUTTON,
-      step: IDX_STEP.PIV_IDP,
-      dataSe: 'piv-card-button',
-      classes: `${piv?.className || ''} piv-button`,
-      variant: 'secondary',
-      Icon: SmartCardIconSvg,
-      onClick: (widgetContext: IWidgetContext) => {
-        // To render the PIV view, we have to use a remediation that is provided on initial load
-        // This remediation doesn't allow a network request, so we have to update the transaction
-        // to set the NextStep as the PIV remediation and change the step name to match what
-        // the transaction transformers expect to render PIV View.
-        // NOTE: IDPs and PIV share the same remediation step name, this is why we update the name
-        // for PIV
-        // Additionally, we clear the messages and other remediations from the PIV view
-        // to prevent the widget from displaying elements that are not related to PIV
-        const { setIdxTransaction } = widgetContext;
-        const pivRemediations = transaction.neededToProceed.filter(
-          (remediation) => (remediation.name === IDX_STEP.PIV_IDP),
-        );
-        setIdxTransaction({
-          ...transaction,
-          messages: [],
-          neededToProceed: pivRemediations,
-          availableSteps: pivRemediations as NextStep[],
-          nextStep: pivRemediations.find(({ name }) => name === IDX_STEP.PIV_IDP) as NextStep,
-        });
-      },
-    },
-  };
+  const fastPassButtonElement = getFastPassButtonElement(transaction);
```

Comment:
moved fastpass button here to make it easier to group, order, and insert idp/custom buttons, along with the divider element.